salt/modules/win_task.py

```diff
@@ -621,21 +622,38 @@ def create_task_from_xml(name,
 
     except pythoncom.com_error as error:
         hr, msg, exc, arg = error.args  # pylint: disable=W0633
-        fc = {-2147216615: 'Required element or attribute missing',
-              -2147216616: 'Value incorrectly formatted or out of range',
-              -2147352571: 'Access denied'}
+        error_code = hex(exc[5] + 2**32)
+        failure_code = error_code
+        fc = {'0x80041319L': 'Required element or attribute missing',
+              '0x80041318L': 'Value incorrectly formatted or out of range',
+              '0x80020005L': 'Access denied',
+              '0x80041309L': "A task's trigger is not found",
+              '0x8004130aL': "One or more of the properties required to run this task have not been set",
+              '0x8004130cL': "The Task Scheduler service is not installed on this computer",
+              '0x8004130dL': "The task object could not be opened",
+              '0x8004130eL': "The object is either an invalid task object or is not a task object",
+              '0x8004130fL': "No account information could be found in the Task Scheduler security database for the task indicated",
+              '0x80041310L': "Unable to establish existence of the account specified",
+              '0x80041311L': "Corruption was detected in the Task Scheduler security database; the database has been reset",
+              '0x80041313L': "The task object version is either unsupported or invalid",
+              '0x80041314L': "The task has been configured with an unsupported combination of account settings and run time options",
+              '0x80041315L': "The Task Scheduler Service is not running",
+              '0x80041316L': "The task XML contains an unexpected node",
+              '0x80041317L': "The task XML contains an element or attribute from an unexpected namespace",
+              '0x8004131aL': "The task XML is malformed",
+              '0x0004131cL': "The task is registered, but may fail to start. Batch logon privilege needs to be enabled for the task principal",
+              '0x8004131dL': "The task XML contains too many nodes of the same type",
+              }
         try:
-            failure_code = fc[exc[5]]
+            failure_code = fc[error_code]
         except KeyError:
-            failure_code = 'Unknown Failure: {0}'.format(error)
-
-        log.debug('Failed to create task: %s', failure_code)
+            failure_code = 'Unknown Failure: {0}'.format(error_code)
+        finally:
+            log.debug('Failed to create task: %s', failure_code)
+        return failure_code
```

Comment:
We should probably raise a CommandExecutionError here instead. If the Return type is Bool, it should always be Bool. It shouldn't return 2 different types.

Reply:
Oh, nice. I didn't know it existed.